### ISSUE TITLE
Building and running NG on Ubuntu 16.04

### ISSUE THREAD

**Issue by darealshinji** [nightingale-media-player/nightingale-hacking#379]
After the most recent changes in nightingale-deps I was able to build the deps and NG itself on Ubuntu 16.04 with gcc 5.3.1 (VirtualBox).

When I try to run it, it'll crash with a segmentation fault:

```
(process:2837): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
Segmentation fault (core dumped)
```

A debug version however will run and even play MP3 sound (requires `gstreamer0.10-alsa` to be installed and `gstreamer0.10-fluendo-mp3` to be built from source).

I've attached the dumped core from the non-debug version and the logged terminal output from the debug version:
[core.zip](https://github.com/nightingale-media-player/nightingale-hacking/files/352015/core.zip)
[ng-debug.log.zip](https://github.com/nightingale-media-player/nightingale-hacking/files/352016/ng-debug.log.zip)
##

**Comment by JazzMaster** [comment]
no you shouldnt need source, I have just bin installed, you need to link it into the private dep folder, however.

**Comment by JazzMaster** [comment]
could someone tell me what that assertion means, all my apps are doing it.

**Comment by darealshinji** [comment]
Nightingale works fine by the way if you install it into an Ubuntu 14.04 chroot prefix and run it from there.

**Comment by ilikenwf** [comment]
The deps don't get along very well with gcc 5+ if I recall. Try using the bin deps in conjunction with the Nightingale source, or build the deps with an older gcc if possible from a ppa.

**Comment by darealshinji** [comment]
I was able to create a bundle and an [AppImage](http://appimage.org/) that can be run on Ubuntu 16.04 and 16.10 and probably most other distros too:

[Nightingale-1.13+20151005-linux-x86_64.AppImage](https://drive.google.com/file/d/0BxClEVSW8QykYmRZaEV5eUd0NFU/view?usp=sharing) (make executable and run)
[nightingale-1.13+20151005-linux-x86_64.tar.xz](https://drive.google.com/file/d/0BxClEVSW8Qyka3Z6MEpiR3NubXc/view?usp=sharing)
[Nightingale-1.13+20151005-linux-x86_64-addons.tar.gz](https://drive.google.com/file/d/0BxClEVSW8QykUzFZc05FOGcwc28/view?usp=sharing) (my favorite addons; not a requirement)

**Comment by ilikenwf** [comment]
Very cool!

**Comment by probonopd** [comment]
@darealshinji works for me on Xubuntu 16.04. Can you share the script you used to produce it?

Installation of plugins fails:

```
Nightingale could not install the file at 

http://addon-files.getnightingale.com/xpis/MLyrics-1.3.5.xpi

because: Unexpected installation error
Review the Error Console log for more details.
-203
```

Not sure if this is the cause, but always make sure you are only trying to write in `$HOME` and not at the install location of Nightingale, since it is read-only (like if it was installed in `/usr`.

**Comment by darealshinji** [comment]
@probonopd I didn't use any script, I built it manually. I used the Ubuntu 14.04 build from my PPA: https://launchpad.net/~djcj/+archive/ubuntu/nightingale
As for the plugins, they should always automatically install into `$HOME/.nightingale/<profile>/extensions/`. I was able to install the plugins from my link, so I guess that's a different issue (?).

**Comment by probonopd** [comment]
Thanks @darealshinji, unfortunately the last `trusty` build is over a year old in the ppa. Can you make a newer one?

The ingredients used in your AppImage should not be built on a more recent base system than the oldest base system your AppImage is intended to run on. Some core libaries, such as glibc, tend to break compatibility with older base systems quite frequently, which means that binaries will run on newer, but not on older base systems than the one the binaries were compiled on.

If the binary is compiled on a newer system than the one you are trying to run it on, then you will run into errors like `failed to initialize: /lib/tls/i686/cmov/libc.so.6: version GLIBC_2.11 not found
` which we want to avoid. Hence, please use ingredients built for trusty or earlier as ingredients for the AppImage.

**Comment by darealshinji** [comment]
> unfortunately the last trusty build is over a year old in the ppa. Can you make a newer one?

Nightingale didn't receive many updates lately. The packages are actually on par with the second last commit.

**Comment by darealshinji** [comment]
@probonopd I have written this untested recipe which should create exactly the same AppImage: https://gist.github.com/darealshinji/abcfffb658f1a801ce12a332809b084b

**Comment by gavinmelb** [comment]
I've been using Nightingale (and Songbird) with MLyrics installed for ages, and recently, this usually means sorting out dependency version problems. However, upgrading to Ubuntu 16.04 has been a show stopper. Cant get latest stable version installed, and the nightly build that does install wont install MLyrics. 
Over at the Nightingale forums I recently asked if anyone could use Snappy to make a snap. Someone else recommended using docker. I was somewhat heartened to see someone had made an Appimage.

The version supplied above, 1.13, works well, but still wont install MLyrics, as **probonopd** noted.

So, I did a fresh install of 14.04 and tried the script provided by **darealshinji**. The script starts, but quickly crashes, closing the terminal (and functions.sh appears to do nothing, and instantly crashing if run separately). - perhaps there are some packages that need installing for the scripts to run.

*Are there any prerequisite packages needed to run these scripts, or extra instructions that may help me build an Appimage?
*Or, if anyone builds an Appimage that works with MLyrics, could they share the file (like darealshinji has done with the more recent version of NG)?

Many thanks from a long-term Nightingale/Songbird user!

**Comment by freaktechnik** [comment]
You should be able to just change the max version to something like 99.\* in mlyrics and it should still work fine, as djcj says, it's not a radically different ngale version.

**Comment by probonopd** [comment]
Can anyone see **why** the AppImage supplied above, 1.13, won't install MLyrics?

**Comment by gavinmelb** [comment]
Many thanks to you **freaktechnik**.

I had simply assumed the issue was with the NG version rather than anything to do with the addon. And, I didn't know that the addons could be extracted and edited like that. 

Opened MLyrics in an archive manager, found the file **install.rdf**, and located the end section setting the maximum versions for Songbird and Nightingale that it is installable on. **Changed em:maxVersion="1.12" to 1.13**
ie.
  <RDF:Description RDF:about="rdf:#$qGbMF2"
                   em:id="nightingale@getnightingale.com"
                   em:maxVersion="1.13.*"
                   em:minVersion="1.11.0" />

Seems to work fine with embedded lyrics in .mp3 files. Didn't find those for .flac. No big deal, can re-do those.

I have tested both the stable version MLyrics-stable.xpi and MLyrics-latest.xpi. Both OK.

Again many thanks.
I now have my fav media player working for the forseeable future!

**Comment by Ratief** [comment]
I just wanted to mention that the AppImage does not work on Ubuntu 17.10. It seg faults.

```
$ ./Nightingale.AppImage 
/bin/bash: /tmp/.mount_R6GvmU/lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)

(process:22385): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
Segmentation fault (core dumped)
```

**Comment by darealshinji** [comment]
@Ratief I made a new build, maybe that works: https://drive.google.com/open?id=11rnTF9uIXicS_ckP_ZNm-NP0qF0NZ_B1

**Comment by Ratief** [comment]
I'm afraid that it has the same error (seg fault). If you have any advice on how I can get more information for you I would be happy to try.

Edit: I noticed that you have the libc version in the filename. On Ubuntu 17.10 I have libc6 version 2.26-0ubuntu2. I'm not sure if that matters or not.

**Comment by darealshinji** [comment]
glibc is downwards compatible, so that shouldn't be an issue.
You can extract the AppImage with `./Nightingale-*.AppImage --appimage-extract` and remove all libraries within `squashfs-root` which you think are causing issues (can't think of a better solution for now).

**Comment by m27315** [comment]
I made a Dockerized version a while back, and it works fine for me:

https://github.com/m27315/nightingale-docker

When I was using KDE Plasma on Gentoo, I was able to easily bind it to the media buttons on my keyboard (play, pause, next, prev).  However, I'm struggling to integrate it into my new Ubuntu 20.04 system.  I must launch Nightingale from the terminal, which is not terrible, but I'd much prefer to use the media buttons.  I came here hoping someone discovered a better way to integrate it into their desktop. :smile:
